### PR TITLE
Add log module specification

### PR DESCRIPTION
**Pull request by MadhukaHarith92**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/2347
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+## 2. Logging
+The Ballerina log API provides functions to log at four levels, which are `DEBUG`, `ERROR`, `INFO`, and `WARN`.
+```ballerina
+log:printDebug("debug log");
+log:printError("error log");
+log:printInfo("info log");
+log:printWarn("warn log");
+```
+
+Users can pass any number of key/value pairs, which need to be displayed in the log message. These can be of the `anydata` type including `int`, `string`, and `boolean`.
```
Comment (daneshk) [inline docs/spec/spec.md:32]: Can we mention abt function pointers and stack trace objects here? we don't need to give  
```suggestion
Users can pass any number of key/value pairs, which need to be displayed in the log message. These can be of the `anydata` type`.
```

File: docs/spec/spec.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+log:printError("error log", stackTrace = e.stackTrace(), username = "Alex92", id = 845315);
+```
+
+Optionally, an error can be passed to the functions.
+```ballerina
+error e = error("something went wrong!");
+log:printError("error log with cause", 'error = e, id = 845315, name = "foo");
+```
+
+## 3. Logging with Context
```
Comment (daneshk) [inline docs/spec/spec.md:49]: IMO, this is not a separate section. should be a part of logging

File: docs/spec/spec.md (hunk 1 above)
Comment (MadhukaHarith92) [reply docs/spec/spec.md]: Fixed in fe465f5

File: docs/spec/spec.md (hunk 2 above)
Comment (MadhukaHarith92) [reply docs/spec/spec.md]: Fixed in fe465f5

**Review by daneshk — APPROVED** [review]

**Pull request merged**